pkg/queue/sharedmain/main.go

```diff
@@ -243,52 +243,53 @@ func Main(opts ...Option) error {
 		concurrencyendpoint = queue.NewConcurrencyEndpoint(env.ConcurrencyStateEndpoint, env.ConcurrencyStateTokenPath)
 	}
 
+	type service struct {
+		name string
+		srv  *http.Server
+		tls  bool
+	}
+
+	var srvs []service
+	var drainer *pkghandler.Drainer
+	var mainServer *http.Server
+
 	// Enable TLS when certificate is mounted.
 	tlsEnabled := exists(logger, certPath) && exists(logger, keyPath)
 
-	mainServer, drainer := buildServer(d.Ctx, env, d.Transport, probe, stats, logger, concurrencyendpoint, false)
-	httpServers := map[string]*http.Server{
-		"main":    mainServer,
-		"metrics": buildMetricsServer(protoStatReporter),
-		"admin":   buildAdminServer(d.Ctx, logger, drainer),
-	}
-	if env.EnableProfiling {
-		httpServers["profile"] = profiling.NewServer(profiling.NewHandler(logger, true))
+	if tlsEnabled {
+		mainServer, drainer = buildServer(d.Ctx, env, d.Transport, probe, stats, logger, concurrencyendpoint, true)
+		srvs = append(srvs, service{name: "tlsMain", srv: mainServer, tls: true})
+		srvs = append(srvs, service{name: "tlsAdmin", srv: buildAdminServer(d.Ctx, logger, drainer), tls: true})
+	} else {
```

Comment:
I suspect tls tests fail because activator does not disable http with tls (we still need it).

Reply:
ok, will  bring back `admin`

Reply:
My bad, I remembered wrong, I will add `main` when tlsAdmin is up. This will require some more work to get the drainer to work properly (or we could leave fixing the drainer to another round)